resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/delegates/CookieHeaderDelegate.java

```diff
@@ -16,10 +18,40 @@ public Cookie fromString(String value) throws IllegalArgumentException
       return CookieParser.parseCookies(value).get(0);
    }
 
-   public String toString(Cookie value)
+   public String toString(Cookie cookie)
    {
-      StringBuffer buf = new StringBuffer();
-      ServerCookie.appendCookieValue(buf, 0, value.getName(), value.getValue(), value.getPath(), value.getDomain(), null, -1, false);
-      return buf.toString();
+      if (cookie == null)
+      {
+         throw new IllegalArgumentException(Messages.MESSAGES.paramNull());
+      }
+      StringBuilder b = new StringBuilder();
+      b.append("$Version=").append(cookie.getVersion()).append(";");

```

Comment:
Should only be done if `getVersion` is 1; specifically shouldn't be done if it's 0, or you'll risk breaking the web.